pkg/hintrunner/zero/zerohint_keccak.go

```diff
@@ -111,3 +114,101 @@ func createKeccakWriteArgsHinter(resolver hintReferenceResolver) (hinter.Hinter,
 
 	return newKeccakWriteArgsHint(inputs, low, high), nil
 }
+
+// BlockPermutation hint executes the Keccak block permutation function to a segment of memory
+//
+// `newBlockPermutationHint` takes 2 operanders as arguments
+//   - `keccakStateSizeFelts` is the size of the state array, hardcoded to 25 field elements
+//   - `keccakPtr` is a pointer to the address in memory where to write the result of the permutation
+//
+// `newBlockPermutationHint` reads 25 memory cells starting from `keccakPtr -  25`, and writes
+// the result of the Keccak block permutation in the next 25 memory cells, starting from `keccakPtr`
+func newBlockPermutationHint(keccakStateSizeFelts, keccakPtr hinter.ResOperander) hinter.Hinter {
+	name := "BlockPermutation"
+	return &GenericZeroHinter{
+		Name: name,
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.keccak_utils.keccak_utils import keccak_func
+			//> _keccak_state_size_felts = int(ids.KECCAK_STATE_SIZE_FELTS)
+			//> assert 0 <= _keccak_state_size_felts < 100
+			//
+			//> output_values = keccak_func(memory.get_range(
+			//> 	ids.keccak_ptr - _keccak_state_size_felts, _keccak_state_size_felts))
+			//> segments.write_arg(ids.keccak_ptr, output_values)
+
+			keccakStateSize, err := hinter.ResolveAsUint64(vm, keccakStateSizeFelts)
+			if err != nil {
+				return err
+			}
+
+			keccakWritePtr, err := hinter.ResolveAsAddress(vm, keccakPtr)
+			if err != nil {
+				return err
+			}
+
+			// Should we remove this check?
+			if keccakStateSize >= 100 {
+				return fmt.Errorf("keccakStateSize %v is out range 0 <= keccakStateSize < 100", &keccakStateSize)
+			}
+
+			var readAddr = *keccakWritePtr
+			var offset int16 = int16(keccakStateSize)
+			var negOffset int16 = -offset
+
+			readAddr, err = readAddr.AddOffset(negOffset)
+			if err != nil {
+				return err
+			}
+
+			inputValuesInRange, err := vm.Memory.GetConsecutiveMemoryValues(readAddr, int16(keccakStateSize))
+			if err != nil {
+				return err
+			}
+
+			var keccakInput []uint64
+
+			for _, valueMemoryValue := range inputValuesInRange {
+				valueFelt, err := valueMemoryValue.FieldElement()
+				if err != nil {
+					return err
+				}
+				valueUint256 := uint256.Int(valueFelt.Bits())
+				valueUint64 := valueUint256.Uint64()
```

Comment:
Why not use `Uint64()` method for Memory Values here instead? Is it possible the elements you get in `inputValuesInRange` don't fit?

Reply:
i think i was just not aware at the time i could cast it directly from MemoryValue
thanks!

Reply:
indeed its way better with your solution:)